lib/Parse/ParsePattern.cpp

```diff
@@ -849,6 +851,19 @@ void Parser::parseAsyncThrows(
 
     StringRef keyword = Tok.getText();
     throwsLoc = consumeToken();
+    
+    if (!peekToken().isKeyword()) {
+      BacktrackingScope backtrackingScope(*this);
+      if (peekToken().is(tok::kw_throws)) {
+        ASTContext &Ctx = SF.getASTContext();
+        DiagnosticSuppression SuppressedDiags(Ctx.Diags);
+        backtrackingScope.cancelBacktrack();
```

Comment:
Why is this needed?

Reply:
I use the backtrack for searching of my previous word was a keyword and if it was the throws keyword, but maybe there's some other way to do it.

Reply:
I think what you should do is something like:

```c++
bool hasType = false;
{
  BacktrackingScope backtrack(*this);
  hasType = canParseType();
}

if (hasType) {
  ParserResult<TypeRepr> result = parseType();
  throwsType = result.getPtrOrNull();
}
```

Reply:
That is good for consuming the type token! Thank you! But I'm unable to check that the previous token was `tok::throws`